### PR TITLE
MOTS-360: Refactor of Locations Component, fixed states.

### PR DESCRIPTION
**Pull request by pstudzinski**
Refactor of Locations Component, fixed states.
https://applab.atlassian.net/browse/MOTS-360

### CODE REVIEW

File: src/webapp/js/components/locations.js [hunk 1]
```diff
@@ -126,17 +126,18 @@ class Locations extends Component {
     this.handleTabSelect = this.handleTabSelect.bind(this);
   }
 
+  componentWillMount() {
+    this.setState({ selectedIndex: Number.parseInt(this.props.match.params.tabIndex, 10) });
+    if (!this.props.match.params.tabIndex) {
+      this.setState({ selectedIndex: 0 });
```
Comment (pmuchowski) [inline src/webapp/js/components/locations.js]: you don't need to set state to 0, it's the default value

File: src/webapp/js/components/locations.js [hunk 2]
```diff
@@ -126,17 +126,18 @@ class Locations extends Component {
     this.handleTabSelect = this.handleTabSelect.bind(this);
   }
 
+  componentWillMount() {
+    this.setState({ selectedIndex: Number.parseInt(this.props.match.params.tabIndex, 10) });
+    if (!this.props.match.params.tabIndex) {
```
Comment (pmuchowski) [inline src/webapp/js/components/locations.js]: you should change it to if (this.props.match.params.tabIndex) and move this.setState({ selectedIndex: Number.parseInt(this.props.match.params.tabIndex, 10) }); inside this if

File: src/webapp/js/components/locations.js (hunk 1 above)
Comment (pstudzinski) [reply src/webapp/js/components/locations.js]: ok, removed

File: src/webapp/js/components/locations.js (hunk 2 above)
Comment (pstudzinski) [reply src/webapp/js/components/locations.js]: changed

**Pull request merged**